src/wizard-settings/header-bidding-gam/order.js

```diff
@@ -0,0 +1,320 @@
+			setBidders( await apiFetch( { path: '/newspack-ads/v1/bidders' } ) );
+		} catch ( err ) {
+			setError( err );
+		}
+		if ( orderId ) {
+			// Fetch order.
+			try {
+				const data = await apiFetch( {
+					path: `/newspack-ads/v1/bidding/gam/order?id=${ orderId }`,
+					method: 'GET',
+				} );
+				setConfig( {
+					orderId: data.order_id,
+					name: data.order_name,
+					revenueShare: data.revenue_share,
+					bidders: data.bidders,
+				} );
+				setOrder( data );
+			} catch ( err ) {
+				setError( err );
+			}
+			// Fetch LICA config.
+			try {
+				const licaConfig = await fetchLicaConfig( orderId );
+				const batches = Math.ceil( licaConfig.length / lica_batch_size );
+				setTotalBatches( batches );
+				setTotalSteps( 3 + batches );
+			} catch ( err ) {
+				setError( err );
+			}
+		}
+		setInFlight( false );
+	}, [] );
+
+	useEffect( () => {
+		onPending( inFlight );
+	}, [ inFlight ] );
+
+	const createType = async ( type, requestData = { batch: 0, fixing: false } ) => {
+		return await apiFetch( {
+			path: '/newspack-ads/v1/bidding/gam/create',
+			method: 'POST',
+			data: {
+				...requestData,
+				id: config?.orderId || requestData.id || null,
+				type,
+				config: {
+					order_name: config?.name,
+					revenue_share: config?.revenueShare,
+					bidders: config?.bidders,
+				},
+			},
+		} );
+	};
+
+	const create = async ( fixing = false ) => {
+		setError( null );
+		setInFlight( true );
+		let pendingOrder = { ...order };
+		try {
+			if ( ! pendingOrder || ! pendingOrder.order_id ) {
+				setStep( 1 );
+				pendingOrder = await createType( 'order', { fixing } );
+				setOrder( pendingOrder );
+				setConfig( { ...config, orderId: pendingOrder.order_id } );
+			}
+			if ( ! pendingOrder?.line_item_ids?.length ) {
+				setStep( 2 );
+				pendingOrder = await createType( 'line_items', { id: pendingOrder.order_id, fixing } );
+				setOrder( pendingOrder );
+			}
+			const licaConfig = await fetchLicaConfig( pendingOrder.order_id );
+			const batches = Math.ceil( licaConfig.length / lica_batch_size );
+			setTotalBatches( batches );
+			setTotalSteps( 3 + batches );
+			const start = pendingOrder?.lica_batch_count || 0;
+			if ( batches > start ) {
+				for ( let i = start; i < batches; i++ ) {
+					const batch = i + 1;
+					setStep( 2 + batch );
+					pendingOrder = await createType( 'creatives', {
+						id: pendingOrder.order_id,
+						batch,
+						fixing,
+					} );
+					setOrder( pendingOrder );
+				}
+			}
+			setStep( 3 + batches );
+			if ( typeof onSuccess === 'function' ) {
+				await onSuccess( pendingOrder );
+			}
+		} catch ( err ) {
+			if ( orderId || isLastAttempt ) {
+				// Unrecoverable error.
+				if ( typeof onUnrecoverable === 'function' ) await onUnrecoverable( pendingOrder, err );
+				setOrder( null );
+			} else {
+				// Make it fail unrecoverably if it fails on next attempt.
+				if ( pendingOrder?.order_id ) {
+					setLastAttempt( true );
+				}
+				setError( err );
```

Comment:
If creating an order results in an error message, if you go to create a new order the modal still shows the error. We should probably clear the message when starting to create a new order.

<img width="733" alt="Screen Shot 2022-02-18 at 11 59 21 AM" src="https://user-images.githubusercontent.com/2230142/154745624-40ca05c0-7ce9-48f3-8535-aeb0741d989f.png">

Reply:
You're right! 98f80314f5b2c0a67f598448f8f880c3c484365a removes the `error` prop that was being passed to the `<Order />` component.